apps/testing/src/main/java/com/akto/test_editor/Utils.java

```diff
@@ -328,4 +333,92 @@ public static boolean isJsonPayload(String payload) {
+                location, operationMap.getOrDefault("replace_with", "").toString(), operation);
+        } catch (Exception e) {
+            return urlModifierPayload;
+        }
+        return urlModifierPayload;
+    }
+
+    public static String buildNewUrl(UrlModifierPayload urlModifierPayload, String oldUrl) {
+        String url = "";
+        if (urlModifierPayload.getOperationType().equalsIgnoreCase("regex_replace")) {
+            if (urlModifierPayload.getRegex() != null && !urlModifierPayload.getRegex().equals("")) {
+                url = Utils.applyRegexModifier(oldUrl, urlModifierPayload.getRegex(), urlModifierPayload.getReplaceWith());
+            } else {
+                URI uri = null;
+                try {
+                    uri = new URI(oldUrl);
+                    oldUrl = uri.getPath();
+                } catch (Exception e) {
+                    // TODO: handle exception
+                }
+                String[] urlTokens = oldUrl.split("/");
+                Integer position = urlModifierPayload.getPosition();
+                if (position <= 0 || position >= urlTokens.length) {
+                    if (uri != null && uri.getHost() != null) {
+                        return uri.getScheme() + "://" + uri.getHost() + oldUrl;
+                    } else {
+                        return oldUrl;
+                    }
+                    
+                }
+                urlTokens[position] = urlModifierPayload.getReplaceWith();
+                url = String.join( "/", urlTokens);
+                if (uri != null && uri.getHost() != null) {
+                    url = uri.getScheme() + "://" + uri.getHost() + url;
+                }
+            }
+        } else {
+            URI uri = null;
+            try {
+                uri = new URI(oldUrl);
+                oldUrl = uri.getPath();
+            } catch (Exception e) {
+                // TODO: handle exception
+            }
+            String[] urlTokens = oldUrl.split("/");
+            Integer position = urlModifierPayload.getPosition();
+            if (position <= 0 || position > urlTokens.length) {
+                if (uri != null && uri.getHost() != null) {
+                    return uri.getScheme() + "://" + uri.getHost() + oldUrl;
+                } else {
+                    return oldUrl;
+                }
+            }
+
+            String[] newUrlTokens = new String[urlTokens.length];
+            for (int i = 1; i < position; i++) {
+                newUrlTokens[i-1] = urlTokens[i];
+            }
+            newUrlTokens[position - 1] = urlModifierPayload.getReplaceWith();
+            for (int i = position; i < urlTokens.length; i++) {
+                newUrlTokens[i] = urlTokens[i];
+            }
+            url = String.join( "/", newUrlTokens);
+            url = "/" + url;
+            if (uri != null && uri.getHost() != null) {
```

Comment:
lot of duplicated code here. Can you please refactor?

Reply:
cool